src/commands/app/test.js

```diff
@@ -10,40 +10,229 @@ OF ANY KIND, either express or implied. See the License for the specific languag
+    aioLogger.debug(`runExtensionTest:testFolders unit:${unitTestFolder} e2e:${e2eTestFolder}`)
+    aioLogger.debug(`runExtensionTest hooks.test:${extensionConfig.hooks.test}`)
+
+    // if hooks.test available, we run that instead
+    if (extensionConfig.hooks.test) {
+      commandList.push({
+        actionName,
+        type: 'hook',
+        command: extensionConfig.hooks.test
+      })
+    } else if (action) { // filter by ext/action name
+      const commands = await this.filterActions(action, extensionConfig, flags)
+      if (commands.length === 0) {
+        this.log(`No package and action matches action filter(s): ${JSON.stringify(action)}`)
+      }
+      commandList.push(...commands)
+    } else { // run everything
+      if (unit) {
+        commandList.push({
+          actionName,
+          type: 'unit',
+          command: 'jest',
+          args: ['--passWithNoTests', unitTestFolder]
+        })
+      }
+      if (e2e) {
+        commandList.push({
+          actionName,
+          type: 'e2e',
+          command: 'jest',
+          args: ['--passWithNoTests', e2eTestFolder]
+        })
+      }
+    }
+
+    const results = []
+    for (const cmd of commandList) {
+      this.log(chalk.yellow(`${extensionName}/${cmd.actionName} (${cmd.type}) - running tests...`))
+
+      aioLogger.debug(`runExtensionTest:runScript cwd:${extensionConfig.root} cmd:${JSON.stringify(cmd)}`)
+      try {
+        await runScript(cmd.command, extensionConfig.root, cmd.args)
+        results.push({ ...cmd, extensionName, passed: true })
+      } catch (e) {
+        results.push({ ...cmd, extensionName, passed: false })
+        aioLogger.debug(`runExtensionTest:runScript exception:${e.toString()}`)
+      }
+      this.log()
+    }
+
+    return results
+  }
+}
 
 Test.flags = {
-  ...BaseCommand.flags,
-  unit: flags.boolean({
-    char: 'u',
-    description: 'runs unit tests (default).',
-    default: true,
-    exclusive: ['e2e']
+  extension: flags.string({
+    char: 'e',
+    description: 'the extension(s) to test',
+    exclusive: ['action'],
+    multiple: true
+  }),
+  action: flags.string({
+    char: 'a',
+    description: 'the action(s) to test',
+    exclusive: ['extension'],
+    multiple: true
+  }),
+  all: flags.boolean({
+    description: 'run both unit and e2e tests',
+    default: false
   }),
   e2e: flags.boolean({
-    char: 'e',
-    description: 'runs e2e tests.',
-    exclusive: ['unit']
+    description: 'run e2e tests',
+    default: false
+  }),
+  unit: flags.boolean({
+    description: 'run unit tests',
+    default: false
   })
 }
+
+Test.description = `Run tests for an Adobe I/O App
+If no flags are specified, by default only unit-tests are run.
+
+For the --action flag, it tries a substring search on the 'extension-name/action-name' pair for an action.
```

Comment:
hmm I like the idea and the effort however the action flag in `build` and `deploy` works not in the same way it looks for `ow-package-name/ow-action-name` and is a perfect match..

I wonder for consistency, should we postpone this one?

Reply:
or can we keep it ? Do we need to change it build/deploy then ?

Reply:
ah it's actually filtering on `ow-package-name/ow-action-name` (derived from `manifest.full.packages` config. I based the help text off my implementation plan, I will amend it to the correct help text